server/server.go

```diff
@@ -127,113 +120,91 @@ func (s *Server) HandleEvent(ctx context.Context, e lookout.Event) error {
 	return err
 }
 
-// HandleReview sends request to analyzers concurrently
-func (s *Server) HandleReview(ctx context.Context, e *lookout.ReviewEvent, safePosting bool) error {
-	ctx, logger := ctxlog.WithLogFields(ctx, log.Fields{
-		"provider": e.Provider,
-	})
-	logger.Infof("processing pull request")
+func (s *Server) AnalyzeAndComment(ctx context.Context, e lookout.Event, safePosting bool) error {
+	comments, err := s.analyze(ctx, e, safePosting)
+	if err == nil {
+		if err := s.post(ctx, e, comments, safePosting); err != nil {
+			s.status(ctx, e, lookout.ErrorAnalysisStatus)
+			err = fmt.Errorf("posting analysis failed: %s", err)
```

Comment:
This assignment does not apply over [first `err` definition #line-124](https://github.com/src-d/lookout/pull/442/commits/52e1fa8a904ae92161e0dd977e46ed4428d2a723#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR124), but over [its own scope #line-126](https://github.com/src-d/lookout/pull/442/commits/52e1fa8a904ae92161e0dd977e46ed4428d2a723#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR126)

We avoid it, doing early returns; it means, that we stop processing as soon as an error condition is met; example:
```go
if err := doSomething(); err != nil {
    return fmt.Errorf("new deep error") // we stop processing
}
// do other things
```

instead of moving errors and reusing variables containing it, like in your example:
```go
var err error
if err := doSomething(); err != nil { // we shadow the upper err definition
    err = fmt.Errorf("new deep error") // this definition does not apply over the upper scope
}
return err // nothing was overriden; we got <nil>
```

You can see an example about it at: https://play.golang.org/p/I75A6xUtJgN